docs/source/daml-integration-kit/index.rst

```diff
@@ -0,0 +1,329 @@
+
+.. _integration-kit_writing_code:
+
+Writing the Code
+----------------
+
+Once you've completed the above steps, we recommend following these steps to
+implement your own server and validator.
+
+1. Clone our example DAML Ledger backed by an in-memory key-value store from
+   the `digital-asset/daml-on-x-example <https://github.com/digital-asset/daml-on-x-example>`__.
+
+   **TODO: create this example repository**
+
+2. Read the example code jointly with the :ref:`integration-kit_library_overview`
+   given below.
+3. Combine all the knowledge gained to decide on the architecture for your
+   DAML on `X` ledger.
+4. Implement your architecture; and let the world know about it.
+
+Use the feedback form on this documentation page or GitHub issues on the
+`digital-asset/daml <https://github.com/digital-asset/daml>`__ repository to
+get into contact with us. Questions and feedback welcome!
+
+
+.. _integration-kit_library_overview:
+
+Library Infrastructure Overview
+-------------------------------
+
+We provide the following four Scala libraries to help you implement your
+server and validator. They are released as part of the DAML SDK. Changes
+to them are explained as part of the :doc:`/support/release-notes`.
+As explained in :ref:`integration-kit_writing_code`,
+this section is best read jointly with the code in
+`digital-asset/daml-on-x-example <https://github.com/digital-asset/daml-on-x-example>`__.
+
+``participant-state.jar`` (`source code <https://github.com/digital-asset/daml/blob/master/ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/package.scala>`__)
+  contains interfaces abstracting over the state of
+  a participant node relevant for a DAML Ledger API server. These are the
+  interfaces whose implemention is specific to a particular `X` ledger. These
+  interfaces are optimized for ease of implementation.
+``participant-state-index.jar`` (`source code <https://github.com/digital-asset/daml/tree/master/ledger/participant-state-index>`__)
+  contains code for reading the abstract state
+  of a participant node and indexing it to satisfy the read access
+  patterns required for serving the Ledger API. The library provides both
+  an interface enumerating all read access methods and an in-memory
+  reference implementation of that interface.
+  We expect to provide a persistent, SQL-backed index in the future
+  (tracked in this `GitHub issue <https://github.com/digital-asset/daml/issues/581>`__).
+``api-server-damlonx.jar`` (`source code <https://github.com/digital-asset/daml/blob/master/ledger/api-server-damlonx/src/main/scala/com/daml/ledger/api/server/damlonx/Server.scala>`__)
+  contains code that implements a DAML Ledger API
+  server given implementations of the interfaces in ``participant-state.jar``.
+``daml-engine.jar`` (`source code <https://github.com/digital-asset/daml/blob/master/daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala>`__)
```

Comment:
Some readers might be looking for clues how daml engine fits into the overall architecture depicted on the diagram below. Maybe we should show how it fits into the LedgerApiServer.

Reply:
Thx. Added a comment.